File: mario.c
#include <stdio.h>
#include <cs50.h>

// build pyramid 
void build (height) {
   
    
    if (height >= 1)
        {
        printf("       #");
        printf("  ");
        printf("#\n");

        }
    if (height >= 2)
        {
        printf("      ##");
        printf("  ");
        printf("##\n");
        }
    if (height >= 3)
        {
        printf("     ###");
        printf("  ");
        printf("###\n");
        }
    if (height >= 4)
        {
        printf("    ####");
        printf("  ");
        printf("####\n");
        }
    if (height >= 5)
        {
        printf("   #####");
        printf("  ");
        printf("#####\n");
        }
    if (height >= 6)
        {
        printf("  ######");
        printf("  ");
        printf("######\n");
        }
    if (height >= 7)
        {
        printf(" #######");
        printf("  ");
        printf("#######\n");
        }
    if (height >= 8)
        {
        printf("########");
        printf("  ");
        printf("########\n");
        }     
    
}    
  
int get_height(string prompt)
{
    int n;
    do
    {
        n = get_int("%s", prompt);
    }
    while (n < 1 || n > 8);
    return n;
}

int main(void)
{
    int height = get_height("How high can mario jump? ");
    build(height);
}

File: mario.py
from cs50 import get_int


def main():
    height = get_height()
    build(height)


def get_height():
    while True:
        # get size for pyramid from user
        height = get_int("Height: ")
        if 0 < height < 9:
            break
    return height
    
    
def build(height):
    row = 0
    spaces = height - 1
    
    for row in range(height):
        # print spaces
        for s in range(spaces, 0, -1):
            print(" ", end="")
        
        # blocks on the left    
        for b in range(row + 1):
            print("#", end="")
        
        # middle space    
        print("  ", end="")
       
        # blocks on the right
        for b in range(row + 1):
            print("#", end="")
            
        # take out space
        spaces -= 1
        # new line
        print()
        
         
main()